### PR TITLE
Move cli_testing_examples under crates/cli/tests

### LINKED ISSUE #6557: cli_run::expects_dev_and_test fails when the roc repo is cloned into a directory not named "roc"
This test is sensitive to the path the roc repo is cloned into. (Mine is in `~/dev/`.)

```
$  cargo test --package roc_cli --test cli_run -- cli_run --nocapture

...

thread 'cli_run::expects_dev_and_test' panicked at '> expected output to end with:
── EXPECT FAILED in ...roc/roc/crates/cli_testing_examples/expects/expects.roc ─

This expectation failed:

28│      expect words == []
                ^^^^^^^^^^^

When it failed, these variables had these values:

words : List Str
words = ["this", "will", "for", "sure", "be", "a", "large", "string", "so", "when", "we", "split", "it", "it", "will", "use", "seamless", "slices", "which", "affect", "printing"]

[<ignored for tests>:31] x = 42
[<ignored for tests>:33] "Fjoer en ferdjer frieten oan dyn geve lea" = "Fjoer en ferdjer frieten oan dyn geve lea"
[<ignored for tests>:35] "this is line 24" = "this is line 24"
[<ignored for tests>:21] x = "abc"
[<ignored for tests>:21] x = 10
[<ignored for tests>:21] x = (A (B C))
Program finished!

> but instead got:
── EXPECT FAILED in ...dev/roc/crates/cli_testing_examples/expects/expects.roc ─

This expectation failed:

28│      expect words == []
                ^^^^^^^^^^^

When it failed, these variables had these values:

words : List Str
words = ["this", "will", "for", "sure", "be", "a", "large", "string", "so", "when", "we", "split", "it", "it", "will", "use", "seamless", "slices", "which", "affect", "printing"]

[<ignored for tests>:31] x = 42
[<ignored for tests>:33] "Fjoer en ferdjer frieten oan dyn geve lea" = "Fjoer en ferdjer frieten oan dyn geve lea"
[<ignored for tests>:35] "this is line 24" = "this is line 24"
[<ignored for tests>:21] x = "abc"
[<ignored for tests>:21] x = 10
[<ignored for tests>:21] x = (A (B C))
Program finished!

> stderr was:
🔨 Rebuilding platform...
runtime: 1.262ms
', crates/cli/tests/cli_run.rs:310:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test cli_run::expects_dev_and_test ... FAILED
```

It looks like there's some logic to relativize the file path before printing (a), and some logic in the tests to replace the file path in the output with `<ignored for test>` (b), not sure what's broken here.

a) https://github.com/roc-lang/roc/blob/d232a843e44ee0f93ea63b5ad4ec7eae17067956/crates/reporting/src/report.rs#L76
b) https://github.com/roc-lang/roc/blob/d232a843e44ee0f93ea63b5ad4ec7eae17067956/crates/cli/tests/cli_run.rs#L307

### PR DESCRIPTION
**Pull request by jschear**
Fixes #6557. When these fixtures are under `crates/cli`, the file paths in expect failures are property relativized to the CWD.

This touches the benchmark stuff a bit, not sure how to run those on CI.

### CODE REVIEW

**Comment by Anton-4** [comment]
Thanks for working on this @jschear :)

Would it be possible to get rid of cli_testing_examples and place its subfolders directly in crates/cli/tests?
`crates/cli/tests/cli_testing_examples` reads kind of redundant :smile:

**Comment by jschear** [comment]
Heh, great point. Updated.

File: Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,6 @@ exclude = [
   "ci/benchmarks/bench-runner",
   "ci/repl_basic_test",
   # Examples sometimes have Rust hosts in their platforms. The compiler should ignore those.
-  "crates/cli_testing_examples",
```
Comment (jschear) [inline Cargo.toml:36]: `find crates/cli/tests -type f -name "Cargo.toml"` didn't turn up anything, so I think its ok to remove this?

Comment (Anton-4) [reply Cargo.toml:36]: Yeah, they all use zig platforms now, so it's fine to remove that.

File: crates/cli_utils/src/bench_utils.rs [hunk 2]
```diff
@@ -107,7 +107,10 @@ fn bench_cmd<T: Measurement>(
 
 pub fn bench_nqueens<T: Measurement>(bench_group_opt: Option<&mut BenchmarkGroup<T>>) {
     exec_bench_w_input(
-        &file_path_from_root("crates/cli_testing_examples/benchmarks", "nQueens.roc"),
+        &file_path_from_root(
+            "crates/cli/tests/cli_testing_examples/benchmarks",
```
Comment (Anton-4) [inline crates/cli_utils/src/bench_utils.rs:111]: Looks like the paths in this file still need to be updated

Comment (jschear) [reply crates/cli_utils/src/bench_utils.rs]: Good catch, updated.

**Review by Anton-4 — DISMISSED** [review]
Thanks @jschear, looks good :)

**Comment by jschear** [comment]
Looks like something failed with the benchmarks check; I'll have some time to look into it later today.

**Comment by jschear** [comment]
Hmm, hitting a different issue when trying to run the benchmarks locally (maybe I need to run them in a noninteractive shell somehow?):
```
nix-shell-env ❯ nix develop -c ./ci/benchmarks/bench-runner/target/release/bench-runner --check-executables-changed

Doing a test run to verify benchmarks are working correctly and generate executables.

>>bench "branch": "Gnuplot not found, using plotters backend"
s>>bench "branch": "Benchmarking bench-group_wall-time/Benchmarking \"nQueens\""
>>bench "branch": "Benchmarking bench-group_wall-time/Benchmarking \"nQueens\": Warming up for 1.0000 s"
>>bench "branch": "Benchmarking bench-group_wall-time/Benchmarking \"nQueens\": Collecting 10 samples in estimated 1.3578 s (30 iterations)"
>>bench "branch": "Benchmarking bench-group_wall-time/Benchmarking \"nQueens\": Analyzing"
>>bench "branch": "bench-group_wall-time/Benchmarking \"nQueens\""
>>bench "branch": "                        time:   [44.852 ms 44.946 ms 45.053 ms]"
>>bench "branch": "Found 1 outliers among 10 measurements (10.00%)"
>>bench "branch": "  1 (10.00%) high mild"
thread 'main' panicked at 'expected output to end with "396354 & 396354\n" but instead got Out {
    cmd_str: "/Users/jschear/dev/roc/bench-folder-branch/crates/cli/tests/benchmarks/cFold",
    stdout: "Please enter an integer\n",
    stderr: "",
    status: ExitStatus(
        unix_wait_status(
            11,
        ),
    ),
}', crates/cli_utils/src/bench_utils.rs:60:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
>>bench "branch": ""
thread 'main' panicked at 'Error: time-bench execution failed with exit code exit status: 101.
See output above for error info.
Command was:
        "./bench-folder-branch/target/release/deps/time_bench" "--bench" "--noplot"', src/main.rs:164:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

@Anton-4

**Comment by jschear** [comment]
Oh I see, the CI job checks out `main`, invokes the script to prepare a benchmark folder, then checks out this branch and invokes the script to create another folder. Those folders are going to have a different structure (on `main` there will still be `cli_testing_examples` in the path, on this branch there won't be), and the same bench-runner binary isn't handling both formats. @Anton-4 if that makes sense, any tips on how to handle this? Merge through the failure; or update the bench-runner to handle the old layout?

**Comment by Anton-4** [comment]
I'll test this by checking out the current branch twice for the comparison, if that's good we can change one back to main and force the merge after.

**Comment by Anton-4** [comment]
Benchmarks passed :)

**Pull request merged**